# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fs/api/FileSystemSyncAccessHandle.cpp
# Commit: 302fdb94bf7f
# Full Hash: 302fdb94bf7f5cd817b69fc4eaaaf6ff8b62bec8
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2023-02-06 21:36:23
# Regressor Bug: 1809064
# File Overlap Count: 1
# Description:
#   Bug 1809064 - Use managed endpoints for PFileSystemAccessHandle construction; r=dom-storage-reviewers,jari
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D166342
# ==============================================================================

diff -r 100b45bcf622 -r 302fdb94bf7f dom/fs/api/FileSystemSyncAccessHandle.cpp
--- a/dom/fs/api/FileSystemSyncAccessHandle.cpp	Mon Feb 06 15:01:20 2023 +0000
+++ b/dom/fs/api/FileSystemSyncAccessHandle.cpp	Mon Feb 06 15:10:34 2023 +0000
@@ -16,6 +16,7 @@
 #include "mozilla/dom/FileSystemHandleBinding.h"
 #include "mozilla/dom/FileSystemLog.h"
 #include "mozilla/dom/FileSystemManager.h"
+#include "mozilla/dom/FileSystemManagerChild.h"
 #include "mozilla/dom/FileSystemSyncAccessHandleBinding.h"
 #include "mozilla/dom/Promise.h"
 #include "mozilla/dom/UnionTypes.h"
@@ -92,8 +93,8 @@
 
 FileSystemSyncAccessHandle::FileSystemSyncAccessHandle(
     nsIGlobalObject* aGlobal, RefPtr<FileSystemManager>& aManager,
+    mozilla::ipc::RandomAccessStreamParams&& aStreamParams,
     RefPtr<FileSystemAccessHandleChild> aActor, RefPtr<TaskQueue> aIOTaskQueue,
-    mozilla::ipc::RandomAccessStreamParams&& aStreamParams,
     const fs::FileSystemEntryMetadata& aMetadata)
     : mGlobal(aGlobal),
       mManager(aManager),
@@ -121,9 +122,16 @@
 Result<RefPtr<FileSystemSyncAccessHandle>, nsresult>
 FileSystemSyncAccessHandle::Create(
     nsIGlobalObject* aGlobal, RefPtr<FileSystemManager>& aManager,
-    RefPtr<FileSystemAccessHandleChild> aActor,
     mozilla::ipc::RandomAccessStreamParams&& aStreamParams,
+    mozilla::ipc::ManagedEndpoint<PFileSystemAccessHandleChild>&&
+        aAccessHandleChildEndpoint,
     const fs::FileSystemEntryMetadata& aMetadata) {
+  auto accessHandleChild = MakeRefPtr<FileSystemAccessHandleChild>();
+
+  QM_TRY(MOZ_TO_RESULT(
+      aManager->ActorStrongRef()->BindPFileSystemAccessHandleEndpoint(
+          std::move(aAccessHandleChildEndpoint), accessHandleChild)));
+
   QM_TRY_UNWRAP(auto streamTransportService,
                 MOZ_TO_RESULT_GET_TYPED(nsCOMPtr<nsIEventTarget>,
                                         MOZ_SELECT_OVERLOAD(do_GetService),
@@ -134,8 +142,8 @@
   QM_TRY(MOZ_TO_RESULT(ioTaskQueue));
 
   RefPtr<FileSystemSyncAccessHandle> result = new FileSystemSyncAccessHandle(
-      aGlobal, aManager, std::move(aActor), std::move(ioTaskQueue),
-      std::move(aStreamParams), aMetadata);
+      aGlobal, aManager, std::move(aStreamParams), std::move(accessHandleChild),
+      std::move(ioTaskQueue), aMetadata);
 
   auto autoClose = MakeScopeExit([result] {
     MOZ_ASSERT(result->mState == State::Initial);